test/src/test/java/org/neo4j/ogm/autoindex/BaseAutoIndexManagerTestClass.java

```diff
@@ -242,25 +253,13 @@ void executeForConstraints(Consumer<List<ConstraintDefinition>> consumer) {
     void executeCreate(String... statements) {
         for (String statement : statements) {
             logger.info("Execute CREATE " + statement);
-            Result execute = service.execute("CREATE " + statement);
-            execute.close();
+            session.query("CREATE " + statement, emptyMap());
         }
     }
 
     void executeDrop(String... statements) {
         for (String statement : statements) {
-            // need to handle transaction manually because when the service.execute fails with exception
-            // it does not clean up the tx resources, leading to deadlock later
-            Transaction tx = service.beginTx();
-            try {
-                service.execute("DROP " + statement);
-
-                tx.success();
-            } catch (Exception e) {
-                logger.trace("Could not execute drop for statement (this is likely expected) {}", statement, e);
-                tx.failure();
-            }
-            tx.close();
+            session.query("DROP " + statement, emptyMap());
```

Comment:
This was intentionally using the underlying GraphDatabaseService to make the preparation/cleanup more isolated from possible OGM issues. Were there any problems with this?